packages/type-utils/tests/TypeOrValueSpecifier.test.ts

```diff
@@ -0,0 +1,288 @@
+import { parseForESLint } from '@typescript-eslint/parser';
+import type { TSESTree } from '@typescript-eslint/utils';
+import Ajv from 'ajv';
+import path from 'path';
+
+import type { TypeOrValueSpecifier } from '../src/TypeOrValueSpecifier';
+import {
+  typeMatchesSpecifier,
+  typeOrValueSpecifierSchema,
+} from '../src/TypeOrValueSpecifier';
+
+describe('TypeOrValueSpecifier', () => {
+  describe('Schema', () => {
+    const ajv = new Ajv();
+    const validate = ajv.compile(typeOrValueSpecifierSchema);
+
+    function runTestPositive(data: unknown): void {
+      expect(validate(data)).toBe(true);
+    }
+
+    function runTestNegative(data: unknown): void {
+      expect(validate(data)).toBe(false);
+    }
+
+    it.each([['MyType'], ['myValue'], ['any'], ['void'], ['never']])(
+      'matches a simple string specifier',
```

Comment:
[Tests] Without a specifier in the test name, no test failure indicates which case it's actually failing on. Here and elsewhere:

```suggestion
      'matches a simple string specifier: %s',
```